gpkit/nomials/math.py

```diff
@@ -1,25 +1,18 @@
 """Signomial, Posynomial, Monomial, Constraint, & MonoEQCOnstraint classes"""
 import numpy as np
-from .data import simplify_exps_and_cs
-from .nomial_core import Nomial
-from .substitution import substitution, parse_subs
+from .core import Nomial
 from ..constraints import SingleEquationConstraint
-from ..small_classes import Strings, Numbers, Quantity
+from ..small_classes import Strings, Numbers
 from ..small_classes import HashVector
 from ..keydict import KeySet
 from ..varkey import VarKey
 from ..small_scripts import mag
-from .. import ureg
-from .. import DimensionalityError
 from ..exceptions import InvalidGPConstraint
+from .map import NomialMap
+from .substitution import parse_subs
 
 
-def non_dimensionalize(posy):
-    "Non-dimensionalize a posy (warning: mutates posy)"
-    if posy.units:
-        posy.convert_to('dimensionless')
-        posy.cs = posy.cs.magnitude
-        posy.units = None
+EMPTY_EXP = HashVector()
```

Comment:
they are mutable, and though monomials aren't mutated anywhere in the current code this isn't much of a speedup and should probably be removed?